projects/drivers/xilinx_platform/platform_drivers.c

```diff
@@ -180,55 +193,62 @@ int32_t spi_init(struct spi_desc **desc,
 		 const struct spi_init_param *param)
 {
 	spi_desc *descriptor;
+	extra_spi_desc *extra_descriptor;
+	extra_spi_init_param *extra_param;
 	int32_t ret;
 
 	descriptor = (struct spi_desc *) calloc(1, sizeof(*descriptor));
-	if (!descriptor)
+	extra_descriptor = (struct extra_spi_desc *) calloc(1,
```

Comment:
i don't think you need casts when allocating memory. I has no actual value. Check this in other places also.

Reply:
Maybe we should keep the casts for the sake of consistency.

Reply:
There are arguments for keeping the casting (https://www.gnu.org/software/libc/manual/html_node/Basic-Allocation.html), but since we are following the Linux kernel coding guidelines, we shouldn't do it.